src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -779,6 +779,31 @@ def projection_in_str(self) -> str:
             )
         return self._projection_in_str
 
+    def with_subqueries(self, subquery_plans: List[SnowflakePlan]) -> "SelectStatement":
+        """Update pre-actions, post-actions and schema to capture necessary subquery_plans
+        encountered during plan resolution."""
+        for plan in subquery_plans:
+            for query in plan.queries[:-1]:
+                if self.pre_actions is None:
+                    self.pre_actions = []  # pragma: no cover
+                if query not in self.pre_actions:
+                    self.pre_actions.append(query)
```

Comment:
this part of code seems repeated code block with the with_subqueries, and seems missing the API call handling also. can we have a utility function to handle the with_subqueries, maybe move the with_subqueries to a util file, and take a snowflakeplan or selectable as input and return a snowflakeplan or selectable